packages/backend/src/services/UserAttributesService/UserAttributeUtils.ts

```diff
@@ -82,14 +96,24 @@ export const getFilteredExplore = (
                     ...table,
                     metrics: Object.fromEntries(
                         Object.entries(table.metrics).filter(
-                            ([metricName, metric]) =>
-                                !metric.tablesReferences ||
-                                metric.tablesReferences.every(
-                                    (tableReference) =>
-                                        filteredTableNames.includes(
-                                            tableReference,
-                                        ),
-                                ),
+                            ([metricName, metric]) => {
+                                const parentDimensionIsHidden =
+                                    hiddenDimensions.some(
+                                        (dimension) =>
+                                            dimension.sql === metric.sql,
```

Comment:
afaik this is the most reliable  way to know if a metric comes from a dimension, right ? 
![Screenshot from 2024-03-05 09-42-25](https://github.com/lightdash/lightdash/assets/1983672/e4adfe07-5f1a-40fa-a35b-457a67c1f37c)

Reply:
Yes, but won't work for metrics with custom SQL. To support those, the metrics should have a `dimensionReferences: Array<{table: string; name: string;}>;` similar to the `tablesTeferences`.

Reply:
Yeah, I've been trying to figure out a way to link dimensions in metrics, but at least on the exploreCompiler, metrics and dimension are not linked, we might need to look before that happens, perhaps at `convertExplores` on `translator` ... I'll keep looking :eyes: